src/main/java/arb/logic/commands/project/EditProjectCommand.java

```diff
@@ -0,0 +1,176 @@
+package arb.logic.commands.project;
+
+import static arb.logic.parser.CliSyntax.PREFIX_DEADLINE;
+import static arb.logic.parser.CliSyntax.PREFIX_NAME;
+import static arb.model.Model.PREDICATE_SHOW_ALL_PROJECTS;
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+import java.util.Optional;
+
+import arb.commons.core.Messages;
+import arb.commons.core.index.Index;
+import arb.commons.util.CollectionUtil;
+import arb.logic.commands.Command;
+import arb.logic.commands.CommandResult;
+import arb.logic.commands.exceptions.CommandException;
+import arb.model.ListType;
+import arb.model.Model;
+import arb.model.project.Deadline;
+import arb.model.project.Project;
+import arb.model.project.Title;
+
+/**
+ * Edits the details of an existing project in the address book.
+ */
+public class EditProjectCommand extends Command {
+    public static final String COMMAND_WORD = "edit-project";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the project identified "
+            + "by the index number used in the displayed project list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_NAME + "TITLE] "
+            + "[" + PREFIX_DEADLINE + "DEADLINE] "
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_NAME + "Sunset painting "
+            + PREFIX_DEADLINE + "2023-07-05";
+
+    public static final String MESSAGE_EDIT_PROJECT_SUCCESS = "Edited Project: %1$s";
+    public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
+    public static final String MESSAGE_DUPLICATE_PROJECT = "This project already exists in the address book.";
+
+    private final Index index;
+    private final EditProjectCommand.EditProjectDescriptor editProjectDescriptor;
```

Comment:
For this statement and in line 50 (and other instances I may not have caught), I'm fairly certain you can just write `EditProjectDescriptor` for the class.